src/libs/web-preset.ts

```diff
@@ -22,21 +29,48 @@ function isDocumentVisible(): boolean {
 
 const fetcher = (url: any) => fetch(url).then(res => res.json())
 
-function setOnFocus(callback: (...args: unknown[]) => void) {
-  if (!isWindowEventTarget) return
-  window.addEventListener('focus', callback, false)
-  window.addEventListener('visibilitychange', callback, false)
+const invokeFocus = () => invokeEvent(domListenersMap, 'focus')
+const invokeOnline = () => invokeEvent(domListenersMap, 'online')
+
+function useOnFocus(callback: listenerInterface): () => void {
```

Comment:
keep the name ***setOnFoucs*** seems batter since this function is not a custom react hook.